plugin/trino-kudu/src/main/java/io/trino/plugin/kudu/KuduKerberosConfig.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.kudu;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.validation.FileExists;
+
+import javax.validation.constraints.NotNull;
+
+import java.io.File;
+import java.util.Optional;
+
+public class KuduKerberosConfig
+{
+    private String clientPrincipal;
+    private String clientKeytab;
+    private File config;
+    // The kudu client defaults to using "kudu" if this is undefined
+    private Optional<String> kuduPrincipalPrimary = Optional.empty();
+
+    @NotNull
+    public String getClientPrincipal()
+    {
+        return clientPrincipal;
+    }
+
+    @Config("kerberos.client.principal")
```

Comment:
`kudu.authentication.client.principal`

Reply:
I stole the config naming convention from other connectors that have kerberos config. But I can rename these 👍

Reply:
At some point I'd like to extract such configs and use config binder to add a prefix that matches the connector factory name automatically.

It's unwritten (and un-enforced) convention that catalog config properties are prefixed by the connector's name.